internal/command/e2etest/pluggable_state_store_test.go

```diff
@@ -119,3 +121,79 @@ func TestPrimary_stateStore_workspaceCmd(t *testing.T) {
 		t.Errorf("unexpected output, expected %q, but got:\n%s", expectedMsg, stdout)
 	}
 }
+
+// Tests using `terraform state` subcommands in combination with pluggable state storage:
+// > `terraform state show`
+// > `terraform state list`
+func TestPrimary_stateStore_stateCmds(t *testing.T) {
+
+	if !canRunGoBuild {
+		// We're running in a separate-build-then-run context, so we can't
+		// currently execute this test which depends on being able to build
+		// new executable at runtime.
+		//
+		// (See the comment on canRunGoBuild's declaration for more information.)
+		t.Skip("can't run without building a new provider executable")
+	}
+
+	t.Setenv(e2e.TestExperimentFlag, "true")
+	tfBin := e2e.GoBuild("github.com/hashicorp/terraform", "terraform")
+
+	fixturePath := filepath.Join("testdata", "initialized-directory-with-state-store-fs")
+	tf := e2e.NewBinary(t, tfBin, fixturePath)
+
+	workspaceDirName := "states" // see test fixture value for workspace_dir
+
+	// In order to test integration with PSS we need a provider plugin implementing a state store.
+	// Here will build the simple6 (built with protocol v6) provider, which implements PSS.
+	simple6Provider := filepath.Join(tf.WorkDir(), "terraform-provider-simple6")
+	simple6ProviderExe := e2e.GoBuild("github.com/hashicorp/terraform/internal/provider-simple-v6/main", simple6Provider)
+
+	// Move the provider binaries into the correct directory .terraform/providers/ directory
+	// that will contain provider binaries in an initialized working directory.
+	platform := getproviders.CurrentPlatform.String()
+	if err := os.MkdirAll(tf.Path(".terraform/providers/registry.terraform.io/hashicorp/simple6/0.0.1/", platform), os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
+	if err := os.Rename(simple6ProviderExe, tf.Path(".terraform/providers/registry.terraform.io/hashicorp/simple6/0.0.1/", platform, "terraform-provider-simple6")); err != nil {
```

Comment:
Nitpick: It would be nice to factor out the path as a variable as it's quite a long one and it's being repeated here.